move/interchain_token_service/sources/interchain_token_service.move

```diff
@@ -103,15 +103,45 @@ module interchain_token_service::interchain_token_service {
         self.value_mut!(b"disallow_function").disallow_function(version, function_name);
     }
 
+    /// Publicly freeze `CoinMetadata` for a token id and remove it from `CoinInfo` storage.
+    /// Can be used to migrate an existing coin from v0 to v1
+    entry fun migrate_coin_metadata<T>(self: &mut InterchainTokenService, _: &OperatorCap, token_id: address) {
+        self.value_mut!(b"migrate_coin_metadata").migrate_coin_metadata<T>(token_id);
+    }
+
     // ----------------
     // Public Functions
     // ----------------
+
+    // TODO: disable this in version control
     public fun register_coin<T>(self: &mut InterchainTokenService, coin_info: CoinInfo<T>, coin_management: CoinManagement<T>): TokenId {
         let value = self.value_mut!(b"register_coin");
 
         value.register_coin(coin_info, coin_management)
```

Comment:
Make it a one-liner perhaps to match other usage

Reply:
All the `entry` functions are one liner, and all the `public` functions are these two-liners. Do you prefer I streamline everything to one-liners?

Reply:
Created [9747](https://axelarnetwork.atlassian.net/browse/AXE-9747) for this